dispatches/models/nuclear_case/chgtank.py

```diff
@@ -0,0 +1,590 @@
+##############################################################################
+# DISPATCHES was produced under the DOE Design Integration and Synthesis
+# Platform to Advance Tightly Coupled Hybrid Energy Systems program (DISPATCHES),
+# and is copyright (c) 2021 by the software owners: The Regents of the University
+# of California, through Lawrence Berkeley National Laboratory, National
+# Technology & Engineering Solutions of Sandia, LLC, Alliance for Sustainable
+# Energy, LLC, Battelle Energy Alliance, LLC, University of Notre Dame du Lac, et
+# al. All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. Both files are also available online at the URL:
+# "https://github.com/gmlc-dispatches/dispatches".
+#
+##############################################################################
+
+"""
+Basic IDAES unit model for compressed hydrogen gas (chg) tank.
+Uses custom material and energy balance equations.
+Suitable for use with steady state flowsheets indexed with time.
+
+An initial state block (initial_state) is defined
+to pass the intial condition of tank, i.e., Pressure and Temperature.
+For dynamic operations, multiple instances of this model indexed with time
+can be used. For this, equality constraints must be written at
+flowsheet level, i.e., 
+the final state of tank (Pres & Temp) at time (t-1) ==
+    intial state of tank (Pres & Temp) at time (t)
+
+Time window of operation
+
+This property package assumes Ideal gas behaviour of hydrogen gas.
+Adiabatic operations are assumed.
+TO DO: add constraints to enable isothermal operations
+"""
+
+# Import Pyomo libraries
+import pyomo.environ as pyo
+from pyomo.environ import (Var,
+                           Reals,
+                           NonNegativeReals,
+                           Constraint,
+                           units as pyunits)
+# from pyomo.network import Port
+from pyomo.common.config import ConfigBlock, ConfigValue, In
+
+# Import IDAES cores
+from idaes.core import (ControlVolume0DBlock,
+                        declare_process_block_class,
+                        MaterialFlowBasis,
+                        MomentumBalanceType,
+                        UnitModelBlockData,
+                        useDefault)
+from idaes.core.util.config import is_physical_parameter_block
+from idaes.core.util.constants import Constants as const
+from idaes.generic_models.properties.core.eos.ideal import Ideal
+import idaes.logger as idaeslog
+import idaes.core.util.scaling as iscale
+
+__author__ = "Naresh Susarla"
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+def _make_chg_tank_config_block(config):
+    config.declare("dynamic", ConfigValue(
+        domain=In([False]),
+        default=False,
```

Comment:
I think you need to clearly explain how a storage tank can have `dynamic` and `has_holdup` can be `False`. This seems contradictory at first pass, so a comment here is important.